File: algorithms/sorting/insertion-sort/insertion-sort.mjs
/**
 * Return a sorted copy of the input using Insertion Sort.
 * The original array is never mutated.
 */
export function insertionSort(numbers) {
  const array = [...numbers];

  for (let index = 1; index < array.length; index += 1) {
    const key = array[index];
    let position = index;

    while (position > 0 && array[position - 1] > key) {
      array[position] = array[position - 1];
      position -= 1;
    }

    array[position] = key;
  }

  return array;
}

File: algorithms/sorting/insertion-sort/insertion_sort.py
def insertion_sort(numbers):
    """Return a sorted copy of numbers using Insertion Sort."""
    array = numbers.copy()

    for index in range(1, len(array)):
        key = array[index]
        position = index

        while position > 0 and array[position - 1] > key:
            array[position] = array[position - 1]
            position -= 1

        array[position] = key

    return array
